packages/metadata-editor/src/MetadataEditor.tsx

```diff
@@ -87,6 +88,7 @@ export class MetadataEditor extends ReactWidget {
         if (this.name == metadata.name) {
           this.metadata = metadata['metadata'];
           this.displayName = metadata['display_name'];
+          this.requiredFields = metadata['required'];
```

Comment:
There are two levels of 'required' lists, one at the top level - as you have here, and another within the metadata stanza itself - which is probably more what your intention is.  The top level required fields will typically **always** be 'schema_name', 'display_name' and 'metadata' (i.e., system-level properties).  The required fields in the metadata stanza will be specific to what that metadata object is about.
```suggestion
          this.requiredFields = metadata['metadata']['required'];
```
Note, since a metadata object may not require any fields, you'll probably need to check if the `required` key exists in `metadata['metadata']` before attempting to access it.

Reply:
Good catch!